From b0760e6553f06dc85eb759af6412ccde466e4ff4 Mon Sep 17 00:00:00 2001
From: Aleksandr Gumroian <aleksandr.gumroian@nic.cz>
Date: Mon, 16 Jan 2023 14:21:33 +0100
Subject: [PATCH] Fix depracated flask-babel's localselector decorator

Since Flask-Babel 3.0.0 Babel.locale_selector no longer exists.
It must be provided when the Babel() object is created or
when init_app() is called.
---
 reforis/__init__.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/reforis/__init__.py b/reforis/__init__.py
index 3443d9742..46e02cc25 100644
--- a/reforis/__init__.py
+++ b/reforis/__init__.py
@@ -171,12 +171,12 @@ def set_locale(app):
     :param app: Flask (application)
     """
     from flask_babel import Babel
-    babel = Babel(app)

-    @babel.localeselector
     def get_locale():
         return _get_locale_from_backend(app)

+    babel = Babel(app, locale_selector=get_locale)
+
     @app.context_processor
     def add_translations_catalog_to_ctx():
         from flask_babel import get_locale
--
GitLab
